src/common/window/session-storage.js

```diff
@@ -6,34 +6,68 @@
  * @environment Browser script
  */
 import { generateRandomHexString } from '../ids/unique-id'
-import { isBrowserScope } from '../util/global-scope'
 
-export { getCurrentSessionIdOrMakeNew }
-
-const SESS_ID = 'NRBA_SESSION_ID' // prevents potential key collisions in session storage
+const AGENT_PREFIX = 'NRBA/'
 
 /**
- * @returns {string} This tab|window's session identifier, or a new ID if not found in storage
+ * @param {string} key
+ * @param {string} value
+ * @returns true, if item was set successfully. false, if storage is full or setItem threw an exception.
  */
-function getCurrentSessionIdOrMakeNew () {
-  if (!isBrowserScope) {
-    return null
+export function putInBrowserStorage (key, value, subgroup = '') {
+  try {
+    window.sessionStorage.setItem(AGENT_PREFIX + subgroup + key, value)
+    return true
+  } catch (ex) {
+    // Session storage may be full.
+    return false
   }
+}
+/**
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/API/Storage/getItem}
+ * @param {string} key
+ */
+function getFromBrowserStorage (key, subgroup = '') {
+  return window.sessionStorage.getItem(AGENT_PREFIX + subgroup + key)
+}
+export function removeFromBrowserStorage (key, subgroup = '') {
+  return window.sessionStorage.removeItem(AGENT_PREFIX + subgroup + key)
+}
 
+/**
+ * @returns {string} This tab|window's session identifier, or a new ID if not found in storage
+ */
+export function getCurrentSessionIdOrMakeNew () {
   try {
     let sessionId
-    if ((sessionId = window.sessionStorage.getItem(SESS_ID)) === null) {
+    if ((sessionId = getFromBrowserStorage('SESSION_ID')) === null) {
       // Generate a new one if storage is empty (no previous ID was created or currently exists)
       sessionId = generateRandomHexString(16)
-      window.sessionStorage.setItem(SESS_ID, sessionId)
+      putInBrowserStorage('SESSION_ID', sessionId)
     }
     return sessionId
   } catch (ex) {
-    // Session storage quota most likely 0mb and cannot be written to
-    // OR the agent is running in a security context that does not allow
-    // access to the browser sessionStorage
+    // The agent is running in a security context that does not allow access to the sessionStorage or is not in browser window context.
     return null
   }
 }
 
-// In the future, we may expand sessionStorage to, say, auto-save some other agent data between page refreshes.
+/**
+ * Look through the page's sessionStorage items and gather the ones that was set under some grouping label, if applicable.
+ * @param {string} [subgroup] - empty string by default, which will return all items set by browser agent
+ * @returns An object of the key-value items from putInBrowserStorage calls from agents with access to the same storage space.
+ */
+export function getAllStorageItemsOfGroup (subgroup = '') {
+  const prefixFilter = AGENT_PREFIX + subgroup
+  const groupItems = {}
+  try {
+    for (let i = 0; i < window.sessionStorage.length; i++) {
```

Comment:
If this method is called from a worker, it will throw an exception. Should we try to use a storage medium that can be supported from a worker?

Reply:
only called by _configure.js_ which uses `isBrowserScope`